packages/jsii/lib/assembler.ts

```diff
@@ -292,7 +305,10 @@ export class Assembler implements Emitter {
       this._diagnostic(node, ts.DiagnosticCategory.Error, `Could not find module for ${modulePath}`);
       return `unknown.${typeName}`;
     }
-    const fqn = `${pkg.name}.${typeName}`;
+    const submoduleNs = this._namespaceMap.get(type.symbol)?.name;
+    const fqn = submoduleNs == null
+      ? `${pkg.name}.${typeName}`
+      : `${pkg.name}.${submoduleNs}.${typeName}`;
```

Comment:
nit style: I would reverse the logic:

```suggestion
    const fqn = submoduleNs
      ? `${pkg.name}.${submoduleNs}.${typeName}`
      : `${pkg.name}.${typeName}`;
```

Feels more readable to me: "if there is a namespace, then use it, otherwise, dont"